62_CAD/fragment_shader.hlsl

```diff
@@ -372,6 +372,21 @@ float32_t4 calculateFinalColor<true>(const uint2 fragCoord, const float localAlp
     return col;
 }
 
+// TODO: place this MSDF code elsewhere
+float median(float r, float g, float b) {
+    return max(min(r, g), min(max(r, g), b));
+}
+
+float screenPxRange(float2 screenSpaceSizePixels, float2 distanceFieldSizePixels) {
+    float2 unitRange = 1.0 / distanceFieldSizePixels;
+    return max(0.5 * dot(unitRange, screenSpaceSizePixels), 1.0);
```

Comment:
It's correct here that screenPxRange is a single floating point since the aspect ratio remains the same (uniform scaling due to our cropping)

`return max(0.5 * dot(unitRange, screenSpaceSizePixels), 1.0);`
this is wrong on multiple levels, this was originally used to convert distances on both directions to a single one by approximating it. (turning float2 distances into a single distance value)

Now we can compute it with exact precision due to our uniform scaling and cropping

ScreenPxRange should be equal to the ratio of the screenspace size of this texture/glyph to the msdf texture size
BUT remember that screenSpaceSizePixels is the cropped size but MSDFSize(32) is not cropped!

`screenSpaceSizePixels.x/croppedPortionTexelSize.x    ===   screenSpaceSizePixels.y/croppedPortionTexelSize.y`
and croppedPortionTexelSize is `(maxUV-minUV)*MSDFSize`

instead of passing the `float2 screenspace size` from vtx to fragment, just do the screenPxRange computations in vtx shader and pass only a single value `screenSpaceSizePixels.x/((maxUV.x-minUV.x)*MSDFSize.x)`
you probably need the max with 1 when extremely zoomed in.

Reply:
Done